tests/manage/z_cluster/nodes/test_toleration.py

```diff
@@ -0,0 +1,167 @@
+import logging
+import pytest
+
+import re
+
+from ocs_ci.framework.testlib import (
+    tier4,
+    tier4a,
+    E2ETest,
+)
+from ocs_ci.ocs import constants, defaults
+from ocs_ci.ocs.machine import get_labeled_nodes
+from ocs_ci.ocs.resources.pod import (
+    get_all_pods,
+    delete_pods,
+    validate_pods_are_respinned_and_running_state,
+)
+from ocs_ci.ocs.ocp import OCP
+
+
+logger = logging.getLogger(__name__)
+
+
+def check_toleration_on_pods(toleration_key=None):
+    """
+    Function to check toleration on pods
+
+    Args:
+        toleration_key (str): The toleration key to check
+
+    """
+    toleration_key = (
+        toleration_key if toleration_key else "node.ocs.openshift.io/storage"
+    )
+    selector = ["rook-ceph-tools"]
+    pod_objs = get_all_pods(
+        namespace=defaults.ROOK_CLUSTER_NAMESPACE,
+        selector=selector,
+        exclude_selector=True,
+    )
+    for pod_obj in pod_objs:
+        resource_name = pod_obj.name
+        tolerations = pod_obj.get().get("spec").get("tolerations")
+        assert (
+            key for key in tolerations if key["key"] == toleration_key
+        ), f"The toleration {toleration_key} does not exist on {resource_name}"
+        logger.info(f"The Toleration {toleration_key} exists on {resource_name}")
+
+
+def check_taint_on_nodes(taint=None):
+    """
+    Function to check for particular taint on nodes
+
+    Args:
+        taint (str): The taint to check on nodes
+
+    """
+    taint = taint if taint else "node.ocs.openshift.io/storage=true:NoSchedule"
+    obj = OCP()
+    ocs_nodes = get_labeled_nodes(label="cluster.ocs.openshift.io/openshift-storage=")
+    flag = 0
+    for node in ocs_nodes:
+        output = obj.exec_oc_cmd(
+            command=f"describe node {node}", out_yaml_format=False
+        ).split("\n")
+        for line in output:
+            if "Taints:" in line:
```

Comment:
I think that we can init node ojbect for each of this node and directly access needed data from it's get function?  So instead of parsing text output from describe we should processing yaml data provided by that object.